recipes/bx/all/conanfile.py

```diff
@@ -1,81 +1,239 @@
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import MSBuild, VCVars
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.env import VirtualBuildEnv
+from pathlib import Path
 import os
-import glob
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
 
-class BxConan(ConanFile):
+required_conan_version = ">=1.50.0"
+
+
+class bxConan(ConanFile):
     name = "bx"
-    description = "Base library used across multiple projects."
     license = "BSD-2-Clause"
-    topics = ("conan", "bx")
     homepage = "https://github.com/bkaradzic/bx"
     url = "https://github.com/conan-io/conan-center-index"
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
-    settings = "os", "arch", "compiler", "build_type"
-    options = {
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "fPIC": True,
-    }
+    description = "Base library providing utility functions and macros."
+    topics = ("general", "utility")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"fPIC": [True, False], "tools": [True, False]}
+    default_options = {"fPIC": True, "tools": False}
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _bx_folder(self):
+        return "bx"
+
+    @property
+    def _compiler_required(self):
+        return {
+            "gcc": "8",
+            "clang": "3.3",
+            "apple-clang": "5",
+        }
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
-        minimal_cpp_standard = "14"
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, minimal_cpp_standard)
-        minimal_version = {
-            "gcc": "5",
-            "clang": "3.4",
-            "apple-clang": "10",
-            "Visual Studio": "15"
-        }
-        compiler = str(self.settings.compiler)
-        if compiler not in minimal_version:
-            self.output.warn(
-                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
-            self.output.warn(
-                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
-            return
-        version = tools.Version(self.settings.compiler.version)
-        if version < minimal_version[compiler]:
-            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+        if self.settings.os == "Windows":
+            self.lib_ext = "*.lib"
```

Comment:
All of these are exclusively need in the `package()` so it would be nice to move these 🙏 they make a lot of noise from the linter in the diff view